custom-sign-in/app/ProfileScreen.js

```diff
@@ -0,0 +1,169 @@
+﻿/*
+ * Copyright (c) 2019, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import React, {Fragment} from 'react';
+import {
+  SafeAreaView,
+  Button,
+  StyleSheet,
+  Text,
+  View,
+  StatusBar,
+} from 'react-native';
+import {
+  createConfig,
+  authenticate,
+  getAccessToken,
+  EventEmitter,
+} from '@okta/okta-react-native';
+import Spinner from 'react-native-loading-spinner-overlay';
+import configFile from './../oidc.config';
+
+export default class ProfileScreen extends React.Component {
+  static navigationOptions = {
+    title: 'User Profile',
+  };
+
+  constructor() {
+    super();
+    this.state = {
+      authenticated: false,
+      accessToken: null,
+      progress: false,
+    };
+  }
+
+  async componentDidMount() {
+    let self = this;
+    EventEmitter.addListener('signInSuccess', function(e: Event) {
+      self.getAccessToken();
+    });
+    EventEmitter.addListener('onError', function(e: Event) {
+      console.warn(e);
+      self.setState({progress: false});
+    });
+    EventEmitter.addListener('onCancelled', function(e: Event) {
+      console.warn(e);
+      self.setState({progress: false});
+    });
+    await createConfig({
+      clientId: configFile.oidc.clientId,
+      redirectUri: configFile.oidc.redirectUri,
+      endSessionRedirectUri: configFile.oidc.endSessionRedirectUri,
+      discoveryUri: configFile.oidc.discoveryUri,
+      scopes: configFile.oidc.scopes,
+      requireHardwareBackedKeyStore:
+        configFile.oidc.requireHardwareBackedKeyStore,
+    });
+  }
+
+  componentWillUnmount() {
+    EventEmitter.removeAllListeners('signInSuccess');
+    EventEmitter.removeAllListeners('onError');
+    EventEmitter.removeAllListeners('onCancelled');
+  }
+
+  async getAccessToken() {
+    const promise = await getAccessToken();
+    this.setState({accessToken: promise.access_token});
+    this.setState({authenticated: true});
+    this.setState({progress: false});
+  }
+
+  async exchangeSessionToken({sessionToken}) {
+    this.setState({progress: true});
+    authenticate({sessionToken});
+  }
+
+  render() {
+    const {navigation} = this.props;
+    const transaction = navigation.getParam('transaction', 'NO-ID');
+    const firstName = transaction.data._embedded.user.profile.firstName;
+    const lastName = transaction.data._embedded.user.profile.lastName;
+    const login = transaction.data._embedded.user.profile.login;
```

Comment:
nit: Suggest mapping these into a `userProfile` object with all properties exposed. This will help devs see what is all included by default in the profile object:

```
const userProfile = transaction.data._embedded.user.profile;

...

// Access properties directly
<Text style={styles.titleHello}>
  Hello {userProfile.firstName} {userProfile.lastName}
</Text>
```